core/common/src/main/java/alluxio/resource/DynamicResourcePool.java

```diff
@@ -275,6 +276,15 @@ public DynamicResourcePool(Options options) {
         }
       }
     }, options.getInitialDelayMs(), options.getGcIntervalMs(), TimeUnit.MILLISECONDS);
+
+    registerGauges();
+  }
+
+  private void registerGauges() {
+    MetricsSystem.registerGaugeIfAbsent(
+        MetricsSystem.getMetricName("ResourcePoolSize_" + getClass().getName() + "_"
+            + Integer.toHexString(System.identityHashCode(this))),
```

Comment:
This produces a class type + unique class id string as the name for the gauge.

Reply:
I wonder how the hex string will help. For example, if you have `ResourcePoolSize_BlockMasterClientPool_abcdef` and `ResourcePoolSize_BlockMasterClientPool_ghijkl`, how do you further associate them back to the resource user? 

nit: `ResourcePoolSize` -> `ResourcePool`?

Reply:
it probably won't, but provides unique naming at the very least.